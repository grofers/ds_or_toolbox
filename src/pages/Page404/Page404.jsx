import React from "react";
import { Link } from "react-router-dom";
import { Typography } from "@atlas/core";

const Page404 = (props) => {
  const currentPath = props.location.pathname;

  return (
    <main>
      <Typography.Title>Page {currentPath} does not exist</Typography.Title>
      <Typography.Text>
        Go back <Link to="/">Home</Link>
      </Typography.Text>
    </main>
  );
};

export default Page404;
