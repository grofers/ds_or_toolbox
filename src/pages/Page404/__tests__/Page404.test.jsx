import React from "react";
import { render } from "@testing-library/react";
import Page404 from "../Page404";
import { MemoryRouter } from "react-router-dom";

describe("Page404", () => {
  it("should match the snapshot on page load", () => {
    const { asFragment } = render(<Page404 location={{ pathname: "/404" }} />, {
      wrapper: MemoryRouter,
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
