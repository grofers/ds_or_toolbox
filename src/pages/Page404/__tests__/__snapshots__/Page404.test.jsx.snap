// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 should match the snapshot on page load 1`] = `
<DocumentFragment>
  <main>
    <h1
      class="ant-typography"
    >
      Page /404 does not exist
    </h1>
    <span
      class="ant-typography"
    >
      Go back 
      <a
        href="/"
      >
        Home
      </a>
    </span>
  </main>
</DocumentFragment>
`;
